frontend/src/concepts/pipelines/context/PipelinesContext.tsx

```diff
@@ -173,36 +185,149 @@ export const CreatePipelineServerButton: React.FC<CreatePipelineServerButtonProp
+  const { project } = usePipelinesAPI();
 
-  const createCR = () => {
+  const createCR = (config: PipelineServerConfigType) => {
     setCreating(true);
     setError(null);
-    createPipelinesCR(namespace)
-      .then(() => {
-        refreshState();
-        // Don't reset creating state, this component is not needed once we have it and the caller
-        // should navigate away on their own as they will not need to create anymore
-      })
-      .catch((e) => {
-        setCreating(false);
-        setError(e);
-      });
+
+    Promise.all([
+      new Promise<{ key: string; name: string } | undefined>((resolve) => {
+        if (!config.database.useDefault) {
+          const secretName = `pipelines-external-database-password-${project.metadata.name}`;
+          const databaseRecord = config.database.value?.reduce<Record<string, string>>(
+            (acc, { key, value }) => ({ ...acc, [key]: value }),
+            {},
+          );
+
+          createSecret(
+            assembleSecret(project.metadata.name, {
+              [secretName]: databaseRecord[DATABASE_CONNECTION_KEYS.PASSWORD],
+            }),
+          ).then((secret) => {
+            resolve({
+              key: secret.metadata.name,
+              name: secretName,
+            });
+          });
+        }
+        resolve(undefined);
+      }),
+      new Promise<{ secretName?: string; awsData?: AWSDataEntry } | undefined>((resolve) => {
+        if (config.objectStorage.useExisting) {
+          resolve({
+            secretName: config.objectStorage.existingName,
+            awsData: config.objectStorage.existingValue,
+          });
+        } else {
+          const assembledSecret = assembleSecret(
+            project.metadata.name,
+            config.objectStorage.newValue.reduce<Record<string, string>>(
+              (acc, { key, value }) => ({ ...acc, [key]: value }),
+              {},
+            ),
+            'aws',
+          );
+
+          createSecret(assembledSecret).then((secret) => {
+            if (isSecretAWSSecretKind(secret)) {
+              resolve({
+                secretName: assembledSecret.metadata.name,
+                awsData: convertAWSSecretData({
+                  type: DataConnectionType.AWS,
+                  data: secret,
+                }),
+              });
+            }
+            resolve(undefined);
+          });
+        }
+      }),
+    ]).then(([databaseSecret, objectStorageSecret]) => {
+      const awsRecord = objectStorageSecret?.awsData?.reduce<Record<string, string>>(
+        (acc, { key, value }) => ({ ...acc, [key]: value }),
+        {},
+      );
+
+      const databaseRecord = config.database.value?.reduce<Record<string, string>>(
+        (acc, { key, value }) => ({ ...acc, [key]: value }),
+        {},
+      );
+
+      const [externalDBHost, externalDBScheme] =
+        databaseRecord?.[DATABASE_CONNECTION_KEYS.HOST].match(/^(\w+):\/\/(.*+)/) ?? [];
```

Comment:
```suggestion
        databaseRecord?.[DATABASE_CONNECTION_KEYS.HOST].match(/^(\w+):\/\/(.*)/) ?? [];
```
Not sure what this regex is supposed to say... but `.*+` is invalid -- I got an error starting up your PR.